### ISSUE TITLE
Questions on code (ATL)

### ISSUE THREAD

**Issue by anjugopinath** [zhihou7/HOI-CL#10]
Could you answer the below questions please?

1. What do the following keywords in lib/ult/ult.py indicate?

i) Neg_select
ii) pos_h_boxes
iii) neg_h_boxes
iv) pattern_type
v) pattern_channel

2. In which file do you mention the input path for the training dataset? In this case, 
'HOI-CL/Data/hico_20160224_det/images/train2015/'  ?
I executed python tools/Train_ATL_HOCO.py

Thank You.

**Comment by anjugopinath** [comment]
Hi,

In this file:
![image](https://user-images.githubusercontent.com/4654298/131601842-85613e57-dce8-4733-b6f4-49f137cdd48e.png)
The person is interacting with the glass bottle and the pipe. 
There are other bottles, a grater, a scrubber, knife etc. Should I add 'no_interaction' annotations for every item for which there is no interaction? In that case, should I say no_interaction for bottle? Since, there is another bottle that is interacting.

**Comment by zhihou7** [comment]
Yes, for the other bottle (no interaction and no annotation), we currently include this no_interaction in Trainval_Neg_HICO.pkl when we have the object boxes (bottle), that's negative samples.

**Comment by anjugopinath** [comment]
Hi,

Thank You for the reply. Can I train the model if I do not have Trainval_Neg annotations for the new dataset I am using?

**Comment by zhihou7** [comment]
You can train the model. But you might suffer from imbalance/label missing problem.

**Comment by anjugopinath** [comment]
Hi,
Thank You for the reply. 
Can you answer the below questions please?
1.
![image](https://user-images.githubusercontent.com/4654298/133172236-d67055ce-03f8-45e0-823a-a9a49331c944.png)

2. 
![image](https://user-images.githubusercontent.com/4654298/133172273-3a33eb8a-a666-4107-8703-2ef0292d892a.png)

**Comment by zhihou7** [comment]
hoi_to_obj.pkl and hoi_to_verb.pkl store the co-occurence matrix in HICO-DET, i.e. which object and verb are corresponding to a HOI. the name  of the id in  hoi_to_obj.pkl and hoi_to_verb.pkl are provides in hico_list_obj.txt, hico_list_vb.txt, hico_list_hoi.txt respectively. Noticeabley, the id in pkl files starts from 0 while the id in txt files starts from 1.

prior_mask.pkl is similar to hoi_to_obj.pkl, but is for V-COCO. prior_mask.pkl is providd in previous works.

hoi_coco_list_num.txt is not required for training ATL. hoi_coco_list_num.txt just demonstrates the long-tailed distribution.

**Comment by anjugopinath** [comment]
Thank You for your reply. Do I need to create 24_verbs.txt,21_verbs.txt and prior_mask.pkl for training ATL?

**Comment by anjugopinath** [comment]
What is the difference between 
self.num_classes = 600
and
self.compose_num_classes = 600?
in ResNet101_HICO.py inside class ResNet101()

**Comment by zhihou7** [comment]
"num_classes" are annotated class number, while "self.compose_num_classes" is how many types of HOIs you want to compose (that can be larger than self.num_classes or smaller than self.num_classes).

TIN is Transferable Interactiveness knowledge for Human-Object Interaction Detection. The weights aim at balancing the data. it is a traditional re-balance strategy for imbalance data.
